### PR TITLE
Implement missing features to support docs template

### PR DESCRIPTION
**Pull request by yufeih**
#3376 #3417

### CODE REVIEW

**Comment by OsmondJiang** [comment]
> // Copyright (c) Microsoft. All rights reserved.

do we want to whole liquid to move to legacy folder?

---
Refers to: src/docfx/template/liquid/LiquidTemplate.cs:1 in 6a05550. [](commit_id = 6a0555053a85b07da89a6497fe5ebf80c4fcb506, deletion_comment = False)

File: src/docfx/build/Build.cs [hunk 1]
```diff
@@ -40,6 +40,11 @@ public static async Task Run(string docsetPath, CommandLineOptions options, Repo
                 Legacy.ConvertToLegacyModel(docset, context, files, sourceDependencies, tocMap);
             }
 
+            if (!docset.Config.Output.Json && docset.Config.Dependencies.ContainsKey("_themes"))
```
Comment (OsmondJiang) [inline src/docfx/build/Build.cs]: >docset [](start = 17, length = 6)

options.Legacy &&

File: src/docfx/template/Template.cs [hunk 2]
```diff
@@ -25,35 +29,55 @@ internal class Template
...
             var fileMetadata = JObject.FromObject(model.Metadata);
 
             var obj = JObject.FromObject(model);
+            obj["_op_gitContributorInformation"] = new JObject
+            {
+                ["contributors"] = obj["contributors"],
+                ["update_at"] = model.UpdatedAt.ToString(culture.DateTimeFormat.ShortDatePattern, culture),
+                ["updated_at_date_time"] = model.UpdatedAt,
+            };
+            obj["__global"] = _global;
```
Comment (OsmondJiang) [inline src/docfx/template/Template.cs]: don't need do that again, we already did in legacy converter

File: src/docfx/build/Build.cs (hunk 1 above)
Comment (yufeih) [reply src/docfx/build/Build.cs]: Right, this could be merged with LegacyMetadata

File: src/docfx/template/Template.cs [hunk 3]
```diff
@@ -93,5 +139,11 @@ public static string CreateHtmlMetaTags(JObject metadata)
 
             return result.ToString();
         }
+
+        private JObject LoadGlobalTokens(string templateDir)
+        {
+            var path = Path.Combine(templateDir, "LocalizedTokens/docs(en-us).html/tokens.json");
```
Comment (superyyrrzz) [inline src/docfx/template/Template.cs]: >en-us [](start = 71, length = 5)

question: this varies with loc templates. How to support this?

Comment (yufeih) [reply src/docfx/template/Template.cs]: just change it to `/docs({locale}).html` when we support loc

**Comment by yufeih** [comment]
Everything seems good locally:

image

**Pull request merged**